File: Print first n Fibonacci Numbers - GFG/print-first-n-fibonacci-numbers.java
//{ Driver Code Starts
//Initial Template for Java


/*package whatever //do not write package name here */

import java.io.*;
import java.util.*;
class GFG {
	public static void main (String[] args) {
	    
	    //taking input using Scanner class
		Scanner sc=new Scanner(System.in);
		
		//taking total count of testcases
		int t=sc.nextInt();
		
		while(t-->0)
		{
		    //taking total number of elements
		    int n=sc.nextInt();
		    
		    //calling printFibb() method
		    long[] res = new Solution().printFibb(n);
		    
		    //printing the elements of the array
		    for (int i = 0; i < res.length; i++)
		        System.out.print (res[i]+" ");
		    System.out.println();
		}
	}
}
// } Driver Code Ends


//User function Template for Java


class Solution
{
    //Function to return list containing first n fibonacci numbers.
	public static long[] printFibb(int n) 
	{
	    //creating a list to store the numbers. 
	    long[] res = new long[n];
	    int ind = 0;
	    
	    //storing base values for a and b.
        long a=1,b=1;
        
        //pushing 1 once in the list if n>=1 and again if n>=2.
        if(n>=1)
            res[ind++] = 1;
        if(n>=2)
            res[ind++] = 1;
            
         
        for(int i = 2 ; i < n ; ++ i )
        {
            //pushing sum of a and b in the list.
            res[ind++] = a + b;
            long c=a+b;
            
            //updating a as b and b as c (sum of a and b).
            a=b;
            b=c;
        }
        
        //returning the list.
        return res;
    }
}

File: Print first n Fibonacci Numbers - GFG/print-first-n-fibonacci-numbers.js
//{ Driver Code Starts
//Initial Template for javascript
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let dp = new Array();
    dp.push(BigInt(0));
    dp.push(BigInt(1));
    dp.push(BigInt(1));
    for(let i=4;i<=100;i++)
    {
        dp.push(BigInt(0));
    }
    
    let t = parseInt(readLine());
    for(let i=0;i<t;i++)
    {
        let input_line = readLine().split();
        let N = BigInt(input_line[0]);
        let obj = new Solution();
        let ans = obj.printFibb(N, dp);
        let s = "";
        for(let j=0;j<ans.length;j++)
        {
            s+=ans[j];
            s+=" ";
        }
        console.log(s);
    }
}
// } Driver Code Ends


//User function Template for javascript

/**
 * @param {BigInt} N
 * @returns {BigInt[]} 
*/
class Solution 
{
    //Function to return list containing first n fibonacci numbers.
    printFibb(N)
    {
        //creating a list to store the numbers. 
        let ans = new Array();
        
        //storing base values for a and b.
        let a= BigInt(1), b= BigInt(1);
        
        //pushing 1 once in the list if n>=1 and again if n>=2.
        if(N>=1)
            ans.push(BigInt(1));
        if(N>=2) 
            ans.push(BigInt(1));
            
      
        for(let i = 2 ; i < N ; ++ i )
        {
            //pushing sum of a and b in the list.
            ans.push(a+b);
            let c=a+b;
            
            //updating a as b and b as c (sum of a and b).
            a=b;
            b=c;
        }
        
        //returning the list.
        return ans;
    }
}